src/meltano/core/utils/__init__.py

```diff
@@ -645,3 +645,31 @@ def get_no_color_flag() -> bool:
         True if the NO_COLOR environment variable is set to a truthy value, False otherwise.
     """
     return get_boolean_env_var("NO_COLOR")
+
+
+# TODO: Replace with `str.removeprefix` once Meltano drops Python 3.8 support
+def remove_prefix(string: str, prefix: str) -> str:
+    """Remove the specified prefix if it is present.
+
+    Args:
+        string: The string from which the prefix will be removed if present.
+        prefix: The prefix to remove from the string.
+
+    Returns:
+        The given string, with the prefix removed if it was present.
+    """
+    return string[len(prefix) :] if string.startswith(prefix) else string
+
+
+# TODO: Replace with `str.removesuffix` once Meltano drops Python 3.8 support
+def remove_suffix(string: str, suffix: str) -> str:
+    """Remove the specified suffix if it is present.
+
+    Args:
+        string: The string from which the suffix will be removed if present.
+        suffix: The suffix to remove from the string.
+
+    Returns:
+        The given string, with the suffix removed if it was present.
+    """
+    return string[: -len(suffix)] if string.endswith(suffix) else string
```

Comment:
Are these used anywhere?

Reply:
They were, and then while refactoring some stuff I no longer needed them. I'll remove them since they aren't used anywhere, but I bet I'll end up adding them back in before we drop Python 3.8 support :laughing: